lib/sentry/finch_client.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Sentry.FinchClient do
+  @behaviour Sentry.HTTPClient
+
+  @moduledoc """
+  The built-in HTTP client.
+
+  This client implements the `Sentry.HTTPClient` behaviour.
+
+  It's based on the [finch](https://github.com/sneako/finch) Erlang HTTP client,
+  which is an *optional dependency* of this library. If you wish to use another
+  HTTP client, you'll have to implement your own `Sentry.HTTPClient`. See the
+  documentation for `Sentry.HTTPClient` for more information.
+
+  Finch is built on top of NimblePool. If you need to set other pool configuration options,
+  see "Pool Configuration Options" in the source code for details on the possible map values.
+  [finch configuration options](https://github.com/sneako/finch/blob/main/lib/finch.ex)
```

Comment:
We shouldn't point people to source code, that can change (especially `main`). Let's link to Finch for docs